src/validation/schema.js

```diff
@@ -154,6 +157,65 @@ const unwrapType = type => {
     : type
 }
 
+const gatherLocalTypes = defs =>
+  defs
+    .filter(
+      def =>
+        def.kind === 'ObjectTypeDefinition' ||
+        def.kind === 'EnumTypeDefinition' ||
+        def.kind === 'InterfaceTypeDefinition',
+    )
+    .map(def => def.name.value)
+
+const gatherForeignTypes = defs =>
```

Comment:
I've reworked the terminology in the RFC a little, let's call these types `imported types` and this function `gatherImportedTypes` now.